lib/resque/worker.rb

```diff
@@ -861,7 +862,11 @@ def perform_with_fork(job, &block)
 
       begin
         @child = fork do
-          unregister_signal_handlers if term_child
+          if term_child
+            unregister_signal_handlers
+          else
+            @signal_handler.reopen
```

Comment:
Added https://github.com/resque/resque/pull/1499/commits/885d51fb86651be61bdbb4485bbe7b27df405331, but sleep 0.1 * 10 times would be feasible...